### ISSUE TITLE
[CRASH] Exception on server startup

### ISSUE THREAD

**Issue by Totobird-Creations** [samolego/MobDisguises#9]
```
Exception in thread "main" java.lang.RuntimeException: Failed to setup Fabric server environment!
	at net.fabricmc.loader.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:58)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: An exception occurred when launching the server!
	at net.fabricmc.loader.launch.server.FabricServerLauncher.setup(FabricServerLauncher.java:116)
	at net.fabricmc.loader.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:56)
Caused by: java.lang.RuntimeException: An exception occurred when launching the server!
	at net.fabricmc.loader.launch.server.FabricServerLauncher.launch(FabricServerLauncher.java:70)
	at net.fabricmc.loader.launch.server.FabricServerLauncher.setup(FabricServerLauncher.java:114)
	... 1 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.launch.server.FabricServerLauncher.launch(FabricServerLauncher.java:68)
	... 2 more
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:236)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotServer.main(KnotServer.java:28)
	... 7 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	... 9 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
	at net.minecraft.class_2246.<clinit>(class_2246.java:103)
	at net.minecraft.class_3523.<clinit>(class_3523.java:13)
	at net.minecraft.class_5471.<clinit>(class_5471.java:11)
	at net.minecraft.class_5458.method_30573(class_5458.java:44)
	at net.minecraft.class_5458.method_30566(class_5458.java:75)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:723)
	at net.minecraft.class_5458.<clinit>(class_5458.java:74)
	at net.minecraft.class_2378.<clinit>(class_2378.java:266)
	at net.minecraft.class_2966.method_12851(class_2966.java:44)
	at net.minecraft.server.Main.main(Main.java:92)
	... 14 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Callback method onEntitySpawn(Lnet/minecraft/class_1297;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfoReturnable;)V in mobdisguises.common.mixins.json:ServerWorldMixin_SpawnDisguiser failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap mobdisguises-common-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:408)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1291)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
	... 30 more
```

What is this and how do I fix it?

**Comment by James103** [comment]
1. What Minecraft version are you using?
2. Do you have the full mod list? If so, can you please send it here?

**Comment by toto-bird** [comment]
1. 1.17
2.
- Adorn
- Appleskin
- Armor Stand Editor
- Beacon Overhaul
- Bedspreads
- Blockus
- Camp Fires Cook Mobs
- Carpet Extra
- Carpet TIS Addition
- Carved Melons
- Cave Biomes API
- Chain Suspension
- Chest Building Blocks
- Culinaire
- Dawn API
- Day Dream
- Dehydration
- DisguiseLib
- Dual Riders
- Dual Wielding
- Endermen Are Friends
- EnvironmentZ
- Expanded Axe Enchanting
- Extra Origins
- Fabric API
- Carpet
- Fabric Language Kotlin
- Seasons
- Fat Experience Orbs
- Genesis
- Keep Head Names
- Lithium
- MiniMOTD
- MobDisguises
- Mob Origins
- Mo Colours
- More Messages
- More Villagers
- Mo Structures
- Neutral Animals
- Noteblock Expansion
- NPC Variety
- On Soul Fires
- Origins
- Origins Classes
- Parry
- Pehkui
- Performant
- Pettable
- Shulker Box Tooltip
- Sit
- Stackable Potions
- Too Many Origins
- TRansliterationLib
- Universal Ores
- Villager Names
- Simple Voice Chat
- Wandering Collector

I think that's it.

**Comment by James103** [comment]
What happens if you retry without MobDisguises installed? Do you get a similar crash?

**Comment by toto-bird** [comment]
Nope. That was the first thing I tried. No crash without MobDisguises.

**Comment by toto-bird** [comment]
I think figured out the problem. I found the mod on curseforge and it said 1.16. It's not released for 1.17 yet.

**Comment by James103** [comment]
I get the same crash as well.
Mod list:
```
Mods:
  [❌] carpet-extra-1.17.1-1.4.43.jar (disabled)
  [✔️] disguiselib-1.0.5-fabric
  [✔️] fabric-api-0.37.2+1.17
  [❌] fabric-carpet-1.17.1-1.4.44+v210714.jar (disabled)
  [✔️] mobdisguises-1.0.2-fabric
  [❌] notenoughcrashes-3.4.0+1.17-fabric.jar (disabled)
[03:08:25] [main/INFO]: Loading for game Minecraft 1.17.1
```
<details>

<summary> Crash log with just DisguiseLib, Fabric API, and MobDisguises enabled </summary>

```
[03:08:25] [main/INFO]: Loading for game Minecraft 1.17.1
[03:08:26] [main/INFO]: [FabricLoader] Loading 50 mods:
	- disguiselib@1.0.5
	- fabric@0.37.2+1.17
	- fabric-api-base@0.3.0+a02b446318
	- fabric-api-lookup-api-v1@1.2.0+2b5c62d018
	- fabric-biome-api-v1@3.1.11+c345aea818
	- fabric-blockrenderlayer-v1@1.1.5+a02b446318
	- fabric-command-api-v1@1.1.3+5ab9934c18
	- fabric-commands-v0@0.2.2+92519afa18
	- fabric-containers-v0@0.1.12+a02b446318
	- fabric-content-registries-v0@0.2.4+2e8bd82f18
	- fabric-crash-report-info-v1@0.1.5+be9da31018
	- fabric-dimensions-v1@2.0.11+6cefd57718
	- fabric-entity-events-v1@1.1.0+a02b446318
	- fabric-events-interaction-v0@0.4.9+a722d8c018
	- fabric-events-lifecycle-v0@0.2.1+92519afa18
	- fabric-game-rule-api-v1@1.0.7+6cefd57718
	- fabric-item-api-v1@1.2.4+a02b446318
	- fabric-item-groups-v0@0.2.10+b7ab612118
	- fabric-key-binding-api-v1@1.0.4+a02b446318
	- fabric-keybindings-v0@0.2.2+36b77c3e18
	- fabric-lifecycle-events-v1@1.4.4+a02b446318
	- fabric-loot-tables-v1@1.0.4+a02b446318
	- fabric-mining-levels-v0@0.1.3+92519afa18
	- fabric-models-v0@0.3.0+a02b446318
	- fabric-networking-api-v1@1.0.13+2e8bd82f18
	- fabric-networking-blockentity-v0@0.2.11+a02b446318
	- fabric-networking-v0@0.3.2+92519afa18
	- fabric-object-builder-api-v1@1.10.9+b7ab612118
	- fabric-object-builders-v0@0.7.3+a02b446318
	- fabric-particles-v1@0.2.4+a02b446318
	- fabric-registry-sync-v0@0.7.10+e2961fee18
	- fabric-renderer-api-v1@0.4.4+5f02c96918
	- fabric-renderer-indigo@0.4.8+a02b446318
	- fabric-renderer-registries-v1@3.2.0+a02b446318
	- fabric-rendering-data-attachment-v1@0.1.5+a02b446318
	- fabric-rendering-fluids-v1@0.1.13+a02b446318
	- fabric-rendering-v0@1.1.2+92519afa18
	- fabric-rendering-v1@1.7.0+2e8bd82f18
	- fabric-resource-loader-v0@0.4.8+a00e834b18
	- fabric-screen-api-v1@1.0.4+155f865c18
	- fabric-screen-handler-api-v1@1.1.8+a02b446318
	- fabric-structure-api-v1@1.1.13+5ab9934c18
	- fabric-tag-extensions-v0@1.1.4+a02b446318
	- fabric-textures-v0@1.0.6+a02b446318
	- fabric-tool-attribute-api-v1@1.2.12+b7ab612118
	- fabric-transfer-api-v1@1.0.0+3fe3c3f518
	- fabricloader@0.11.6
	- java@16
	- minecraft@1.17.1
	- mobdisguises@1.0.2
[03:08:26] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.2 Source=file:/C:/Users/james/Documents/MultiMC/libraries/net/fabricmc/sponge-mixin/0.9.4+mixin.0.8.2/sponge-mixin-0.9.4+mixin.0.8.2.jar Service=Knot/Fabric Env=CLIENT
[03:08:26] [main/INFO]: Compatibility level set to JAVA_16
[03:08:26] [main/WARN]: @Mixin target net/minecraft/class_2474$class_5124 is public in fabric-tag-extensions-v0.mixins.json:MixinObjectBuilder and should be specified in value
Failed to start Minecraft:
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:236)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)
	... 8 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	... 10 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:162)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:154)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:1111)
	at net.minecraft.class_2246.<clinit>(class_2246.java:103)
	at net.minecraft.class_3523.<clinit>(class_3523.java:13)
	at net.minecraft.class_5471.<clinit>(class_5471.java:11)
	at net.minecraft.class_5458.method_30573(class_5458.java:44)
	at net.minecraft.class_5458$$Lambda$363/0x0000000046684810.get(Unknown Source)
	at net.minecraft.class_5458.method_30566(class_5458.java:75)
	at net.minecraft.class_5458$$Lambda$371/0x0000000046696cc0.accept(Unknown Source)
	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:723)
	at net.minecraft.class_5458.<clinit>(class_5458.java:74)
	at net.minecraft.class_2378.<clinit>(class_2378.java:266)
	at net.minecraft.class_2966.method_12851(class_2966.java:44)
	at net.minecraft.client.main.Main.main(Main.java:139)
	... 15 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Callback method onEntitySpawn(Lnet/minecraft/class_1297;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfoReturnable;)V in mobdisguises.common.mixins.json:ServerWorldMixin_SpawnDisguiser failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap mobdisguises-common-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:408)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1291)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1042)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:395)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:320)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:345)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:569)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:351)
	... 33 more
Exiting with -1
Process exited with code -1.
```

</details>
